packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -143,83 +139,77 @@ const AmountValue = ({
       color={amountValueColor}
       lineHeight={amountLineHeights[type][size]}
     >
-      {value}
+      {amount.formatted}
     </BaseText>
   );
 };
 
-// This function rounds a number to a specified number of decimal places
-// and floors the result.
-export const getFlooredFixed = (value: number, decimalPlaces: number): number => {
-  const factor = 100 ** decimalPlaces;
-  const roundedValue = Math.floor(value * factor) / factor;
-  return Number(roundedValue.toFixed(decimalPlaces));
-};
-
-export const addCommas = (amountValue: number, currency: Currency, decimalPlaces = 0): string => {
-  // If the currency is 'INR', set the locale to 'en-IN' (Indian English).
-  // Otherwise, set the locale to 'en-US' (U.S. English).
-  const locale = currency === 'INR' ? 'en-IN' : 'en-US';
-  return amountValue.toLocaleString(locale, { minimumFractionDigits: decimalPlaces });
-};
-/**
- * This function returns the humanized amount
- * ie: for INR 2000 => 2K
- * for MYR 2000000 => 2M
- */
-export const getHumanizedAmount = ({
-  value,
-  currency,
-  denominationPosition = 'right',
-}: {
-  value: number;
-  currency: Currency;
-  denominationPosition?: 'left' | 'right';
-}): string => {
-  let amountValue = value;
-  const abbreviations = getCurrencyAbbreviations(currency);
-  const abbreviation = abbreviations.find((abbr) => amountValue >= abbr.value);
-
-  if (abbreviation) {
-    amountValue = amountValue / abbreviation.value;
-    const formattedAmountValue = getFlooredFixed(amountValue, 2);
-
-    if (denominationPosition === 'right') {
-      return `${addCommas(formattedAmountValue, currency)}${abbreviation.symbol}`;
-    }
-
-    return `${abbreviation.symbol}${addCommas(formattedAmountValue, currency)}`;
-  }
-
-  return amountValue.toString();
-};
-
 type FormatAmountWithSuffixType = {
   suffix: AmountProps['suffix'];
   value: number;
-  currency: Currency;
-  denominationPosition?: 'left' | 'right';
 };
 
+/**
+ * Returns a parsed object based on the suffix passed in parameters
+ * === Logic ===
+ * value = 12500.45 
+ * if suffix === 'decimals' => {
+    "formatted": "12,500.45",
+    "integer": "12,500",
+    "decimal": ".",
+    "fraction": "45",
+    "isPrefixSymbol": false,
+    "rawParts": [{"type": "integer","value": "12"},{"type": "group","value": ","},{"type": "integer","value": "500"},{"type": "decimal","value": "."},{"type": "fraction","value": "45"}]
+}
+ * else if suffix === 'humanize' => { formatted: "1.2T" }
+ * else => { formatted: "1,23,456" }
```

Comment:
Shouldn't `currency` prop also internally handle default locale?

E.g. if I write 

```js
<Amount currency="INR" value={120000} /> 
```

It should print `INR 1,20,000` right?

Currently it prints `INR 120,000` unless locale is set correctly as per currency

Reply:
@saurabhdaware 
No, the currency shouldn't define the default locale or formatting of the amount string. 
It has to be driven from user's browser locale settings. We also conducted a case study on Stripe for the same, you can find the document over [here](https://docs.google.com/document/d/1ImrtEX3H_QQp4Zb06Dousmmu1hOdsb9lUQPbrP_3YzA/edit#heading=h.6jn10buryynj). (attached in PR desc too)

Your locale in your browser must be `en`, `en-US` or `en-GB` which is why you're seeing amount being printed as `INR 120,000`. You can check the settings from here chrome://settings/languages#lang

Reply:
A currency can have multiple locale, it reads the locale from your browser, in your case it must be `en-US` instead of `en-IN`.

<img width="706" alt="Screenshot 2024-02-07 at 3 43 23 PM" src="https://github.com/razorpay/blade/assets/46647141/ac7afa17-a98c-417b-beb9-54f4ef173fc1">

Reply:
> A currency can have multiple locale, it reads the locale from your browser, in your case it must be en-US instead of en-IN.

Yes yes I meant, Do we know that all Amount instances on page should always be formatted on page-level or browser-configuration-level locale? or should Amount itself also have locale configuration to have consistent formatting

Reply:
> Amount itself also have locale configuration to have consistent formatting

The locale should be set via the i18n SDK via the consumers, if they don't set it will use browsers default which I believe is fine.

> In places like Pricing page or Landing pages, we might look for a consistent formatting of numbers according to currency no irrespective of the page language?

For such cases, the locale can be set by the consumers, right?

Reply:
+1 to @snitin315 here. 
@saurabhdaware The consumers have the option to set the locale state and all of our APIs will internally pull that locale to drive formatting. We fallback to browser settings incase they don't set it. 

Giving an additional `locale` prop to Amount component was ruled out in our initial discussions because of 2 primary reasons 
1. It increases the complexity of props in Amount component. This data should rather come from a global state. 
2. Multiple Amount instances can then have their own unique formatting which will not be consistent with the page. 

lmk if this is clear, else we can jump over a short call and close this. 
// @kamleshchandnani

Reply:
Makes sense. If a usecase comes where we want different Amount instances on page to be differently formatted, then we can come back and explore locale prop option.